python/pyproject.toml

```diff
@@ -24,7 +24,63 @@ requires = [
     # continue using oldest-support-numpy.
     "oldest-supported-numpy>=0.14; python_version<'3.9'",
     "numpy>=1.25; python_version>='3.9'",
-    "setuptools_scm",
-    "setuptools >= 40.1.0",
+    # configuring setuptools_scm in pyproject.toml requires
+    # versions released after 2022
+    "setuptools_scm[toml]>=8",
+    "setuptools>=64",
     "wheel"
 ]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "pyarrow"
+dynamic = ["version"]
+requires-python = ">=3.8"
+dependencies = [
+    "numpy >= 1.16.6"
+]
+description = "Python library for Apache Arrow"
+readme = {file = "README.md", content-type = "text/markdown"}
+license = {text = "Apache Software License"}
+classifiers  = [
+    'License :: OSI Approved :: Apache Software License',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Programming Langauge :: Python :: 3.12',
+]
+maintainers = [
+    {name = "Apache Arrow Developers", email = "dev@arrow.apache.org"}
+]
+
+[project.urls]
+Homepage = "https://arrow.apache.org/"
+Documentation = "https://arrow.apache.org/docs/python"
+Repository = "https://github.com/apache/arrow"
+Issues = "https://github.com/apache/arrow/issues"
+
+[project.optional-dependencies]
+test = [
+    'pytest',
+    'hypothesis',
+    'cffi',
+    'pytz'
```

Comment:
The current test dependency extra in setup.py also includes pandas, maybe keep that here as well? Or do we only want the strictly required ones? (which is maybe fine, but then I think pytz can be removed as well?)

Reply:
I wasn't sure if we only wanted the strictly required ones! I support adding `pandas`. I thought `pytz` was required, but now I see that it is `importorskip`. =) I'll add `pandas`.